tests/test-components/build.rs

```diff
@@ -30,19 +35,42 @@ fn main() {
             .env("RUSTFLAGS", rustflags())
             .env("CARGO_TARGET_DIR", &out_dir);
         eprintln!("running: {cargo:?}");
-        let status = cargo.status().unwrap();
+        let status = cargo.status().expect("`cargo build` failed");
         assert!(status.success(), "{status:?}");
-        eprintln!("{status:?}");
         let const_name = to_shouty_snake_case(&package);
-        let binary_name = manifest.package.unwrap().name.replace('-', "_");
-        let wasm = out_dir
+        let package_name = manifest.package.expect("manifest has no package").name;
+        let binary_name = package_name.replace(['-', '.'], "_");
+        let wasm_path = out_dir
             .join("wasm32-wasi")
             .join("debug")
             .join(format!("{binary_name}.wasm"));
 
+        let adapter_version = package.split('v').last().and_then(|v| match v {
+            "0.2.0-rc-2023-11-10" => Some("15.0.1"),
+            _ => None,
+        });
+
+        if let Some(adapter_version) = adapter_version {
+            let module_bytes = std::fs::read(&wasm_path).expect("failed to read wasm binary");
+            let adapter_bytes = std::fs::read(format!("adapters/{adapter_version}.reactor.wasm"))
+                .expect("failed to read adapter wasm binary");
+            let new_bytes = wit_component::ComponentEncoder::default()
+                .validate(true)
+                .module(&module_bytes)
+                .expect("failed to set wasm module")
+                .adapter("wasi_snapshot_preview1", &adapter_bytes)
+                .expect("failed to apply adapter")
+                .encode()
+                .expect("failed to encode component");
+            std::fs::write(&wasm_path, new_bytes).expect("failed to write new wasm binary");
+        }
+
         // Generate const with the wasm binary path
-        generated_code += &format!("pub const {const_name}: &str = {wasm:?};\n");
-        name_to_path.insert(package, wasm);
+        generated_code += &format!(
+            "pub const {const_name}: &str = \"{}\";\n",
+            wasm_path.display()
```

Comment:
```suggestion
            "pub const {const_name}: &str = {wasm_path:?};\n"
```

:shrug:

Reply:
As noted in https://github.com/fermyon/spin/issues/2158, using the debug implementation does the wrong thing in Windows.

Reply:
Are these tests actually going to be runnable in windows? :grimacing:

Reply:
I guess the service-less ones ought to be...

Reply:
The Python ones should also be runnable, and if they're not that's a bug. I believe the docker containers should also be runnable on Windows, but that's not something I have too much experience with so we'll have to see.

Reply:
lul - actually Windows failed because we _weren't_ using the debug output... Switched back.